### PR TITLE
Adding a socket free SlimService.  Remove all hassles with port conflicts and network setup

### PR DESCRIPTION
**Pull request by six42**
Port number 1 has now a special meaning: Don’t open a socket. Instead communicate via StdIn and StdOut. Nevertheless all output from the SUT is captured, tunneled via StdErr and added to the Execution Log.

This implementation is now the default behavior (= port number 1 is the default) 
The change is fully backward compatible by specifying any other port number.

The Slim protocol number has been increased to 0.5

As there is no need to open a socket it now also easier to implement a Slim Server in other programming languages.
 
The socket functionality is still useful for distributed testing and should be preserved also in the future. Last but not least it is required for backward compatibility.

### CODE REVIEW

**Comment by amolenaar** [comment]
I like it that you can use FitNesse without sockets. This will rule out a whole class of errors. Kudos!

From the PR it was not clear to me how log messages on stdout and stderr mix with the Slim output.

File: src/fitnesse/slim/SlimSlaveSocket.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+  public SlimSlaveSocket() throws IOException {
+
+    // preserve original streams
+    this.stdout = System.out;
+    this.stderr = System.err;
+    this.stdin = System.in;
+
+    // bind System.stdout/System.stderr to original stderr
+    System.setOut(new PrintStream(new LoggingOutputStream(this.stderr, "SOUT"),
```
Comment (six42) [inline src/fitnesse/slim/SlimSlaveSocket.java]: @amolenaar  
This redirects all messages send to stdout to stderr. So stdout is free. SLIM can send results without the SUT mixing in any log messages.
In line 58 in the "accept" method stdout is bound as communication channel for SLIM

**Comment by amolenaar** [comment]
I am trying to use the SocketFactory to also create the pipe-like socket. Expect a PR soon ;)

**Comment by six42** [comment]
Thanks. All merged.

**Comment by amolenaar** [comment]
Awesome. I merged the PR.

**Pull request merged**